### ISSUE TITLE
`rank2` lr initializer not reproducible between 0.4.3 and 0.4.4

### ISSUE THREAD

**Issue by giovp** [ott-jax/ott#483]
**Describe the bug**
I have been trying to reproduce some results using the `rank2` lr initializer without success. Here's the tests I run

**To Reproduce**

**LRFGW**

**Data and modules**
<details>
<summary>Details</summary>

```python
import jax
import jax.numpy as jnp

import matplotlib.pyplot as plt
import numpy as np

from ott.geometry import pointcloud
from ott.problems.quadratic import quadratic_problem
from ott.solvers.quadratic import gromov_wasserstein
from jax import config

import ott
print(ott.__version__)

config.update("jax_enable_x64", True)

def create_points(rng: jax.Array, n: int, m: int, d1: int, d2: int):
    rngs = jax.random.split(rng, 5)
    x = jax.random.uniform(rngs[0], (n, d1))
    y = jax.random.uniform(rngs[1], (m, d2))
    a = jax.random.uniform(rngs[2], (n,))
    b = jax.random.uniform(rngs[3], (m,))
    a = a / jnp.sum(a)
    b = b / jnp.sum(b)
    z = jax.random.uniform(rngs[4], (m, d1))
    return x, y, a, b, z


rng = jax.random.PRNGKey(0)
n, m, d1, d2 = 10_000, 15_000, 200, 300
x, y, a, b, z = create_points(rng, n, m, d1, d2)


geom_xx = pointcloud.PointCloud(x)
geom_yy = pointcloud.PointCloud(y)
geom_xy = pointcloud.PointCloud(x, z)
prob = quadratic_problem.QuadraticProblem(
    geom_xx,
    geom_yy,
    geom_xy=geom_xy,
    a=a,
    b=b,
    fused_penalty=1.0,
)
```

</details>

**`ott 0.4.3`**
```python
init = "random" # or "rank2"
solver = gromov_wasserstein.GromovWasserstein(rank=100,initializer=init)
ot_gwlr = solver(prob)
```

<img width="622" alt="image" src="https://github.com/ott-jax/ott/assets/25887487/7a3f83b2-208c-49fa-b888-953fce03a36d">
<img width="624" alt="image" src="https://github.com/ott-jax/ott/assets/25887487/57f0da6a-4e04-4e71-9bbb-6eb09e07d6be">

**`ott 0.4.4`**
```python
init = "random" # or "rank2"
solver = gromov_wasserstein_lr.LRGromovWasserstein(rank=100,initializer=init)
ot_gwlr = solver(prob)
```

<img width="629" alt="image" src="https://github.com/ott-jax/ott/assets/25887487/d47d9bb9-995f-45ae-943e-f03c9deed777">
<img width="621" alt="image" src="https://github.com/ott-jax/ott/assets/25887487/3a9e6c5d-acb5-4a69-9380-e12ce4308968">

same happens in ott 0.4.5 and also for LR Sinkhorn (see below)

<img width="617" alt="image" src="https://github.com/ott-jax/ott/assets/25887487/75dd137c-9172-47b0-a315-8fd010c68234">
<img width="638" alt="image" src="https://github.com/ott-jax/ott/assets/25887487/e1175f7e-1db9-4ad8-bdd3-c1f725d48178">
<img width="625" alt="image" src="https://github.com/ott-jax/ott/assets/25887487/164da78a-7eae-4c3c-871d-acae8b301f3c">
<img width="636" alt="image" src="https://github.com/ott-jax/ott/assets/25887487/3a75859e-5ba7-473b-b517-c0a71f67ce35">

**Comment by michalk8** [comment]
Hi @giovp , thanks for spotting this! I think it has more to do with the refactoring of LR Sinkhorn/GW, will look into this and see what went wrong.
In terms of quantitative results, do you see worse results in terms of the OT maps?

**Comment by giovp** [comment]
thanks for taking a look @michalk8 !

> In terms of quantitative results, do you see worse results in terms of the OT maps?
yes, as in costs and OT maps are constants throughout iterations

**Comment by giovp** [comment]
I think this can be closed as was fixed in #494
